dspace-spring-rest/src/main/java/org/dspace/app/rest/repository/CommunityRestRepository.java

```diff
@@ -60,18 +70,34 @@ public CommunityRestRepository() {
     protected CommunityRest createAndReturn(Context context) throws AuthorizeException {
         HttpServletRequest req = getRequestService().getCurrentRequest().getHttpServletRequest();
         ObjectMapper mapper = new ObjectMapper();
-        CommunityRest communityRest = null;
+        CommunityRest communityRest;
         try {
             ServletInputStream input = req.getInputStream();
             communityRest = mapper.readValue(input, CommunityRest.class);
         } catch (IOException e1) {
             throw new UnprocessableEntityException("Error parsing request body: " + e1.toString());
         }
 
-        Community community = null;
+        Community community;
+
 
         try {
-            community = cs.create(null, context);
+            Community parent = null;
+            if (StringUtils.isNotBlank(req.getParameter("parentCommunity"))) {
+
+                UUID owningCommunityUuid = UUIDUtils.fromString(req.getParameter("parentCommunity"));
```

Comment:
Again rename variables & error logs here.  "owningCommunity" should be "parentCommunity"